invenio/modules/knowledge/models.py

```diff
@@ -39,9 +40,26 @@ class KnwKB(db.Model):
     __tablename__ = 'knwKB'
     id = db.Column(db.MediumInteger(8, unsigned=True), nullable=False,
                    primary_key=True, autoincrement=True)
-    name = db.Column(db.String(255), server_default='', unique=True)
-    _description = db.Column(db.Text, nullable=False, name="description")
+    _name = db.Column(db.String(255), server_default='',
+                      unique=True, name="name")
+    _description = db.Column(db.Text, nullable=False,
+                             name="description", default="")
     _kbtype = db.Column(db.Char(1), nullable=True, default='w', name="kbtype")
+    slug = db.Column(db.String(255), unique=True, nullable=False, default="")
+    # Enable or disable the access from REST API
+    is_api_accessible = db.Column(db.Boolean, default=True, nullable=False)
+
+    @db.hybrid_property
+    def name(self):
+        """Get name."""
+        return self._name
+
+    @name.setter
+    def name(self, value):
```

Comment:
It looks to me that the slug is generated every time the name is changed. I think this not desirable, since the slug is pretty close to a primary key that you want to keep stable. I think it's only when the slug is empty that you should try to generate it, and likely I would just try one time and if it fails simply throw an exception.